---
title: RNA-seq analysis with Galaxy
type: training
provider: ISU
hideprovider: true
description: In this workshop, participants will work through a complete RNA-seq analysis using SCINet’s Galaxy interface.

multiday: Oct 6 & 8
end-date: 2025-10-08
time: 1 – 5 PM ET
registration:
    url: https://forms.office.com/g/HT6C9ad6CL
tags: bioinformatics galaxy


ubnav:
  - title: Pre-Workshop Instructions
    url: '#pre-workshop-instructions'
---

In this workshop, participants will work through a complete RNA-seq analysis using SCINet’s Galaxy interface.<!--excerpt--> Participants will learn how to create workflows in Galaxy to:  
* Upload and process RNA-seq data.
* Perform quality control, alignment, and expression quantification.
* Identify differentially expressed genes using DESeq2.  

We will also explore ways in which you can share your history and workflows with other SCINet users in Galaxy and how it can be a powerful tool for collaborations.

## Pre-Workshop Instructions: 
To help minimize technical issues and delays at the start of the workshop, please try the following tests prior to the workshop. 

* **Logging on to [Ceres Open OnDemand (OOD)](http://ceres-ood.scinet.usda.gov/):** 
  * Please confirm you can successfully log in to Ceres OOD with your SCINet account [(see OOD login instructions here)](/guides/access/web-based-login). If you are successful, you will be able to see the Ceres OOD home page.   
* **Ceres Shell Access:** 
  * When on Ceres OOD, click on the top navigation bar: “Clusters” > “Ceres Shell Access”. A new tab will appear that looks like a shell terminal (e.g., like PowerShell). 
  * Please confirm you do not receive any error messages or requests to re-authenticate and that the final line looks like “[firstname.lastname@ceres ~]$”.  
* **Logging on to [Galaxy](https://galaxy.scinet.usda.gov):**
  * Please confirm you can successfully log in to Galaxy on SCINet with your SCINet account [(see Galaxy login instructions here)](/guides/application/galaxy#logging-on). If you are successful, you will be able to see the standard Galaxy home screen. 
* **Importing data to Galaxy**
  * While on Ceres Shell Access, navigate to your Galaxy folder in your home directory on Ceres
    
    {: .copy-code }
    ```
    cd /home/$USER/galaxy
    ```
  * Copy workshop data to your Galaxy folder on Ceres by copying the commands below and running them in the Ceres terminal.
    
    {: .copy-code }
    ```
    cp /90daydata/shared/galaxy_workshop/GCF_000001735.4_TAIR10.1_genomic.fna . 
    ```

    {: .copy-code }
    ```
    cp /90daydata/shared/galaxy_workshop/GCF_000001735.4_TAIR10.1_genomic.gtf .
    ```
    
    {: .copy-code }
    ```
    cp /90daydata/shared/galaxy_workshop/*.gz . 
    ```
  * After copying the files to your Galaxy folder, return to the Galaxy tab in your web browser. (If you are not already logged on to Galaxy, please see “Logging on to Galaxy”, above.)  
  * You will import files into Galaxy by clicking on “Upload” in the left panel of the Galaxy home screen. The following download/upload screen should appear:  
    ![Galaxy upload data modal](/assets/img/events/bioinfo/galaxy/upload_modal.png)
  * Click on “Choose remote files.” This will show you the files in your Galaxy folder on Ceres.  
    ![Modal displaying ist of remote files on Ceres](/assets/img/events/bioinfo/galaxy/remote_files.png)
  * Select the files you want to transfer.
    ![Modal with queued files](/assets/img/events/bioinfo/galaxy/remote_queue.png)
  * Click “Start” and wait for the files to be imported into Galaxy. The files will appear in the History pane on the right of your home screen. 

**For more information on importing data to Galaxy, see the [SCINet Galaxy user guide](/guides/application/galaxy#importing-data-to-galaxy).** 
