---
title: RNA-seq analysis with Galaxy
type: training
provider: ISU
hideprovider: true
description: In this workshop, participants will work through a complete RNA-seq analysis using SCINet’s Galaxy interface.

multiday: Jan 26 & 28
end-date: 2026-01-28
time: 1 – 5 PM ET
registration:
    text: Join the waitlist
    url: https://forms.office.com/g/HT6C9ad6CL
tags: bioinformatics galaxy

subnav:
  - title: Pre-Workshop Instructions
    url: '#pre-workshop-instructions'

---

In this workshop, participants will work through a complete RNA-seq analysis using SCINet’s Galaxy interface.<!--excerpt--> Participants will learn how to create workflows in Galaxy to:  
* Upload and process RNA-seq data.
* Perform quality control, alignment, and expression quantification.
* Identify differentially expressed genes using DESeq2.  

We will also explore ways in which you can share your history and workflows with other SCINet users in Galaxy and how it can be a powerful tool for collaborations.

**At this time, registration is closed as we have reached maximum capacity for all workshops. However, you may [complete the registration form](https://forms.office.com/g/HT6C9ad6CL) to be added to our waitlist for future offerings.**

## Pre-Workshop Instructions: 
To help minimize technical issues and delays at the start of the workshop, please try the following tests prior to the workshop. 

* **Logging on to [Ceres Open OnDemand (OOD)](http://ceres-ood.scinet.usda.gov/):** 
  * Please confirm you can successfully log in to Ceres OOD with your SCINet account [(see OOD login instructions here)](/guides/access/web-based-login). If you are successful, you will be able to see the Ceres OOD home page.   
* **Ceres Shell Access:** 
  * When on Ceres OOD, click on the top navigation bar: “Clusters” > “Ceres Shell Access”. A new tab will appear that looks like a shell terminal (e.g., like PowerShell). 
  * Please confirm you do not receive any error messages or requests to re-authenticate and that the final line looks like “[firstname.lastname@ceres ~]$”.  
* **Logging on to [Galaxy](https://galaxy.scinet.usda.gov):**
  * Please confirm you can successfully log in to Galaxy on SCINet with your SCINet account [(see Galaxy login instructions here)](/guides/application/galaxy#logging-on). If you are successful, you will be able to see the standard Galaxy home screen. 
* **Importing data to Galaxy**
  * While on Ceres Shell Access, navigate to your Galaxy folder in your home directory on Ceres
    
    {: .copy-code }
    ```
    cd /home/$USER/galaxy
    ```
  * Copy workshop data to your Galaxy folder on Ceres by copying the commands below and running them in the Ceres terminal.
    
    ```
    cp /90daydata/shared/galaxy_workshop/GCF_000001735.4_TAIR10.1_genomic.fna . 
    cp /90daydata/shared/galaxy_workshop/GCF_000001735.4_TAIR10.1_genomic.gtf .
    cp /90daydata/shared/galaxy_workshop/*.gz . 
    ```
    {: .copy-code }

  * After copying the files to your Galaxy folder, return to the Galaxy tab in your web browser. (If you are not already logged on to Galaxy, please see “Logging on to Galaxy”, above.)  
  * You will import files into Galaxy by clicking on “Upload” in the left panel of the Galaxy home screen. The following download/upload screen should appear:  
    ![Galaxy upload data modal](/assets/img/events/bioinfo/galaxy/upload_modal.png)
  * Click on “Choose remote files.” This will show you the files in your Galaxy folder on Ceres.  
    ![Modal displaying ist of remote files on Ceres](/assets/img/events/bioinfo/galaxy/remote_files.png)
  * Select the files you want to transfer.
    ![Modal with queued files](/assets/img/events/bioinfo/galaxy/remote_queue.png)
  * Click “Start” and wait for the files to be imported into Galaxy. The files will appear in the History pane on the right of your home screen. 

**For more information on importing data to Galaxy, see the [SCINet Galaxy user guide](/guides/application/galaxy#importing-data-to-galaxy).** 



## Tutorial: RNA-Seq Data Analysis with Galaxy
*by Siva Chudalayandi and Lavida Rogers*

RNA-seq experiments are designed to comprehend transcriptomic changes in organisms in response to a certain treatment. They are also designed to understand the cause and/or effect of a mutation by measuring the resulting gene expression changes. Robust algorithms specifically designed to map short stretches of nucleotide sequences to a genome while being aware of the process of RNA splicing have led to many advances in RNAseq analysis. An overview of RNA-seq analysis is summarized in Figure 1.

### Overview
**insert Figure 1: Overview of the RNAseq workflow**

This tutorial  will guide you through a basic RNAseq analysis with Galaxy, beginning at quality checking of the RNA-seq reads through to differential gene expression analysis. 

1. Quality Control 
2. Read alignment 
3. Abundance estimation
4. Differential gene expression analysis
5. Visualization and biological interpretation

We have downloaded an *Arabidopsis* dataset from NCBI for this purpose. Check the [BioProject](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA348194) page for more information.

### Background Information: Experimental design

This experiment compares wild type (WT) and *atrx-1* mutant to analyze how the loss of function of ATRX protein results in changes in gene expression. The **ATRX** protein is a histone chaperone known to be an important player in the regulation of gene expression. RNA was isolated from three WT replicates and three mutant replicates. The transcriptome was enriched/isolated using the plant RiboMinus kit for obtaining total RNA. RNA-seq was carried out on Illumina Hiseq 2500. The sequencing reads are paired-end data, hence we have 2 files per replicate.

{% include table content="| Condition | replicate 1 | replicate 2 | replicate 3 |
| --- | --- | --- | --- |
| WT | SRR4420293_1.fastq.gz <br> SRR4420293_2.fastq.gz | SRR4420294_1.fastq.gz <br> SRR4420294_2.fastq.gz | SRR4420295_1.fastq.gz <br> SRR4420295_2.fastq.gz |
| atrx-1 | SRR4420296_1.fastq.gz <br> SRR4420296_2.fastq.gz| SRR4420297_1.fastq.gz <br> SRR4420297_2.fastq.gz| SRR4420298_1.fastq.gz <br> SRR4420298_2.fastq.gz |" %}

<ol class="usa-process-list">
<li class="usa-process-list__item" markdown="1">

{:.usa-process-list__heading}
### Preparation

If you missed the pre-workshop help session, see the [pre-workshop instructions](#pre-workshop-instructions) to get started with using Galaxy on SCINet. 

</li> 
<li class="usa-process-list__item" markdown="1">

{:.usa-process-list__heading}
### Setting up your workspace

When you login to Galaxy, your History panel will be empty or contain previous work (if you attended the pre-workshop help session or have used Galaxy before). The Galaxy history is a workspace that stores data, tool outputs, and intermediate results so your work can be reviewed, shared, or rerun at anytime.

#### Create and name a new history
We will begin by creating a new history: 

1. Locate the history panel on the right side of the Galaxy interface, click the `+` to create a new history.
2. Click the history title (default: Unnamed history)
3. Rename the history to `RNA-seq_QCandLogs`
4. Save changes

We will use this history for quality control and preprocessing.

</li> 
<li class="usa-process-list__item" markdown="1">

{:.usa-process-list__heading}
### Importing data to Galaxy

Data can be uploaded to Galaxy from a local machine, from remote sources (URLs and Ceres Galaxy folder) or from a shared history. 

Instructions to upload data to Galaxy via Ceres can be found here: 
[Pre-workshop instructions](https://scinet.usda.gov/events/2026-01-26-rna-seq#pre-workshop-instructions)

For this workshop, we will import data via a shared Galaxy history that we created prior to the workshop.  Here is a link to a Galaxy history with all the files needed for the workshop: 

[History Link](https://galaxy.scinet.usda.gov/u/lavidarogers/h/galaxy-workshop-files-1)

Sharing data via histories is also a great option for working with collaborators.

After clicking the shared history link, you will see "Import this History." Click on it to import the files. Galaxy imports shared histories as new histories. This new history will now contain 12 FASTQ files, the *Arabidopsis* genome (.fa) and the annotation file (.gtf).


**insert screenshot **

Let's rename the imported history to `RNA-seq_analysis`

**insert screenshot **

Why are we working with two histories?
Galaxy does not allow histories to be merged. Instead, datasets can be copied between histories as needed as we move through workflows. Using multiple histories helps keep our main analysis and quality control, logs, and preprocessing steps separate.


#### Job Statuses in Galaxy

To understand the stages of your jobs as we progress through the tutorials, you will need to understand how Galaxy display job statuses. 

- **Grey**: The job is queued. 
- **Yellow**: The job is running.
- **Green**: The job has completed successfully.
- **Red**: The job has failed. 
- **Light Blue**: The job is paused. 

Now that we know what job status is represented by each color used in Galaxy, we can get started!

</li> 
<li class="usa-process-list__item" markdown="1">

{:.usa-process-list__heading}
### Preparing the Data for downstream analysis

While still in the `RNA-seq_analysis` history, we will start by organizing our sequencing data so that read pairing is preserved and sample identity (mutant vs wildtype) is clear. 

This preparation involves renaming the datasets, creating paired dataset lists, and flattening collections as needed.

1. Renaming datasets in Galaxy:

   First, we'll rename the data files so that it is easier to see which files represent mutant or WT samples.
     - In the history panel, locate the 12 FASTQ datasets. 
     - Click on the pencil icon to edit attributes of `SRR4420293_1.fastq.gz` and rename this file to `WT_SRR4420293_1`
     - Save changes. 
     - Rename the remaining files as follows:  
       | Original File Name | New Name |
       | -------------------- | ---------- |
       | SRR4420293_2.fastq.gz | WT_SRR4420293_2 |
       | SRR4420294_1.fastq.gz | WT_SRR4420294_1 |
       | SRR4420294_2.fastq.gz | WT_SRR4420294_2 |
       | SRR4420295_1.fastq.gz | WT_SRR4420295_1 |
       | SRR4420295_2.fastq.gz | WT_SRR4420295_2 |
       | SRR4420296_1.fastq.gz | MUT_SRR4420296_1 |
       | SRR4420296_2.fastq.gz | MUT_SRR4420296_2 |
       | SRR4420297_1.fastq.gz | MUT_SRR4420297_1 |
       | SRR4420297_2.fastq.gz | MUT_SRR4420297_2 |
       | SRR4420298_1.fastq.gz | MUT_SRR4420298_1 |
       | SRR4420298_2.fastq.gz | MUT_SRR4420298_2 |

2. Create Paired Dataset Lists

   The data we are working with are paired-end sequencing reads, so we will create paired dataset lists. Paired lists preserve the relationship between forward and reverse reads.  Galaxy automatically detects paired reads based on file name patterns. 

   Commmon pairing patterns: 
     - _R1 and _R2
     - _1 and _2

   Steps for creating paired dataset lists: 

     1. In the history panel, click marked check box to enable file selection
     2. Select the WT FASTQ files
     3. Click on the down arrow that appears near `6 of 14 selected` and select advanced build list
     4. Select list of paired datasets
     5. Based on the filenames having common pairing patterns, Galaxy will autopair
     6. Save this list as `ATRX_wildtype`
     7. Next, we will repeat the above steps for the MUT files and save the list as `ATRX_mutant`

3. Flatten Paired Lists  
   Some tools such as FASTQC and MultiQC do not require nested pairing structure. Flattening will produce one dataset per sample and ensure forward and reverse reads are not treated separately. 

   Here we will create a flattened collection to simplify the QC steps:

     1. Search for the tool: Flatten Collection 
     2. Select the collections you want to flatten: `ATRX_wildtype`
     3. Run the tool 
     4. Rename the output: `ATRX_wildtype_flattened`
     5. Repeat the above steps to also flatten the mutant dataset collection. 

   At this point, paired datasets have been created and flattened. We will now moves these flattened collections into the QC history we created at the beginning of the tutorial. 

   To make copying easier, Galaxy allows you to view two histories at once, side-by-side.   
   Steps: 
     1. Open the History panel
     2. Click the History options menu (three lines)
     3. Select Show histories side by side
     4. Choose:  
          - Left: `RNA-seq_analysis`
          - Right: `RNA_seq_QCandLogs`

        You should now see both histories. 

        ** insert screenshot **

     5. Drag and drop the two flattened lists to the `RNA_seq_QCandLogs` 
     6. Switch to the `RNA_seq_QCandLogs` history

        ** insert screenshot **


</li> 
<li class="usa-process-list__item" markdown="1">

{:.usa-process-list__heading}
### Quality Control 

We use [FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/), a tool that provides a simple way to do quality control checks on raw sequence data coming from high-throughput sequencing pipelines.

#### FastQC 

1. Go to the Tools panel and search for `FastQC`
2. Set the parameters for FastQC
    - Input files: Raw read data from your current history 
    - Select a flattened collection: `ATRX_wildtype_flattened`
    - Leave all other parameters at their default settings.
3. Run the tool 
4. To view output, click the eye icon on the fastqc output 
5. Repeat for `ATRX_mutant_flattened`


#### MultiQC
After running fastqc on multiple datsets, we can use [MultiQC](https://github.com/MultiQC/MultiQC) to combine all QC reports into a single summary report. This allows you to produce a report that compares results from all samples. 

1. Go to the tools panel and search for multiQC
2. Which tool was used generate logs? Select FASTQC
    - Type of FastQC output: Raw Data 
    - Select the relevant collections (both WT and mutant)

3. Run the tool
4. Both HTML and Stats outputs are produced

You can preview the HTML output in Galaxy and inspect for any problems.



#### Trimming
We will use [Trim Galore](https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/) to trim adapters off the RNAseq reads. This is a wrapper script around `cutadapt`.

1. Go to the Tools Panel and search for Trim galore
2. Parameters:
    - Library: paired-end collection
    - Adapter sequence: Automatic detection
    - Settings: Default
3. Run Tool
4. Output: Trimmed Collection for each input

#### Transition to RNA-seq Analysis History

At this point, reads are analysis-ready. 

Next, we will copy the trimmed FASTQ files into the `RNA_seq_analysis`` history and switch histories.

The `RNA_seq_analysis` history should now contain your trimmed FASTQ files, the reference genome and the GTF annotation file. 

Note: Galaxy does not overwrite data. Every action creates a new dataset, so your original files remain unchanged.

</li> 
<li class="usa-process-list__item" markdown="1">

{:.usa-process-list__heading}
### Mapping to a reference genome

There are several **mapping** programs available for **aligning RNAseq reads to the genome**. Generic aligners such as `BWA`, `bowtie2`, `BBMap`, etc., are not suitable for mapping RNAseq reads because they are not splice-aware. **RNAseq reads are mRNA reads that only contain exonic regions**, hence mapping them back to the genome requires splitting the individual reads that span an intron. It can be done only by splice-aware mappers. In this tutorial, we will use [HISAT2](https://ccb.jhu.edu/software/hisat2/index.shtml), a successor of `Tophat2`.

Steps for using HISAT2 in Galaxy: 
1. Search the tool panel for HISAT2
2. Set your parameters
    - Source for reference genome: Use a genome from history
    - Is this a single or paired library: Paired-end dataset collection (because we created a collected of the raw fastq files)
    - Galaxy tries to automate the steps as much as possible, so because you indicated that the library is a paired-end dataset collection, Galaxy automatically selected the collection in your history. 
    - Paired collection: raw files (a collection in your history)
    - Note to always double check selected inputs even if Galaxy automatically selects the accepted formats. 
    - Specify strand information * reverse (RF)
    - All other parameters can remain set to default settings. 

3. Run Tool 

</li> 
<li class="usa-process-list__item" markdown="1">

{:.usa-process-list__heading}
### Abundance Estimation

For quantifying transcript abundance from RNA-seq data, there are many programs available. Popular tools include [featureCounts](https://academic.oup.com/bioinformatics/article/30/7/923/232889?login=false) and [HTSeq](https://academic.oup.com/bioinformatics/article/38/10/2943/6551247?login=false). We will need a file with aligned sequencing reads (SAM/BAM files generated in previous step) and a list of genomic features (from the GFF file). featureCounts is a highly efficient general-purpose read summarization program that counts mapped reads for genomic features such as genes, exons, promoter, gene bodies, genomic bins and chromosomal locations. It also outputs statistics for the overall summarization results, including the number of successfully assigned reads and the number of reads that failed to be assigned due to various reasons. We can run featureCounts on all SAM/BAM files at the same time or individually.

Steps for running featureCounts in Galaxy: 
1. Search for the tool featurecounts 
2. Set the parameters
    - Alignment file : dataset collection 
    - Specify strand information * RF
    - Gene annotation file: A GFF/GTF file in your history - automatically selects the annotation file, but always double check. 
    - Does the input have read pairs? Yes, paired end, but count them as a single fragment 

3. Run tool 

</li> 
<li class="usa-process-list__item" markdown="1">

{:.usa-process-list__heading}
### Differential Gene Expression Analysis

After read alignment and read counting, differental expression analysis is performed with [DESeq2](https://bioconductor.org/packages//release/bioc/html/DESeq2.html). This identifies genes whose expression differ significantly between factors (mutant vs. wildtype samples)

DESeq2 normalizes count data, estimates dispersion across replicates, and tests for differential expression between defined conditions. In Galaxy, we will use the tool DESeq2 and manually assign samples to experimental conditions. 

1. Search for Deseq2 in the tool panel 
2. Set the tool parameters
    - How: select datasets per level
    - Factor Name: genotype
    - Factor Level: wildtype
    - Count files: 
    - Factor Level: mutant
    - Count files: 
    - Leave others to default settings
3. Run tool 

</li> 
</ol>

## Functional interpretation and reproducible workflows in Galaxy

### Functional enrichment with GOseq
We have identified differentially expressed genes and now we want to biologically interpret what these results mean. Biological interpretation requires asking:

*Are there any biolgical functions or processes that are overrepresented among these genes?*

[Gene ontology (GO)](https://geneontology.org/) enrichment analysis helps us answer this question by testing whether genes associated with specific biological processes (BP), molecular functions (MF), or cellular components (CC) occur more frequently than expected by chance. 

[GOseq](https://bioconductor.org/packages//2.12/bioc/html/goseq.html) is an R-based method designed for handling RNA-seq data. GOseq works by modeling the relationship between gene length and differential gene expression and adjusting enrichment tests to account for bias. GOseq is an R package and in this case, Galaxy will execute R code through the tool GOSeq in the background and return the results as Galaxy datasets.

**Attribution:**
The following steps in this tutorial were adapted from the [Galaxy Training Network Reference-based RNA-seq tutorial](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html Online) with modifications for this workshop.  

### Prepare inputs for GOSeq
To run GOSeq, we need three inputs: 
    1. Differential gene expression results
    2. Gene length information
    3. Gene category mapping (GO terms) (imported from the shared history)

#### Prepare the first input for GOseq
The purpose of this step is to convert the DESeq2 results table into a binary indicator identifying whether each gene is differentially expressed. In this step we will be generating a boolean (TRUE/FALSE) variable based on the adjusted p-value threshold `(p<0.05)`.  

Steps: 
1. Search for the tool "Compute on rows" in the tool panel
2. Set the following parameters:
   - Input file: the DESeq2 result file 
   - Add expression: bool(float(c7)<0.05)
      - *c7: column 7 (adjusted p-value)*
      - *float(c7): allows numeric comparisons*
      - *<0.05: the significant threshold*
      - *bool(...)*: converts to True/False*
   - Mode of the operation: Append
   - Under "Error handling":
      - Autodetect column types: No
      - If an expression cannot be computed for a row: Fill in a replacement value
      - Replacement value: False
3. Run the tool

This creates an additional column in the DESeq2 results table that indicates if a gene is differentially expressed. GOSeq only needs the gene names and the binary indicator, so next we will generate a new dataset with only these two columns. 

Steps: 

1. Search for the tool "Cut"
2. Set parameters:
     - Cut columns: c1,c8
     - Delimited by: Tab
     - From: the output of the Compute tool
3. Run the tool

To ensure all gene names are capitalized, we will change the case of the gene name column by:

1. Search for the tool "Change Case"
2. Set parameters:
      - From: the output of the previous Cut tool
      - Change case of columns: c1 (first column with gene names)
      - Delimited by: Tab
      - To: Upper case
3. Run the tool
4. Rename the output to Gene IDs and differential expression 

#### Prepare the gene length file
We will now extract the gene length information.  

Steps: 
1. Search for the tool "Extract Dataset"
2. Set parameters:
      - Input List: featureCounts on collection N: Feature lengths
      - How should a dataset be selected?: The first dataset
3. Run the tool


Change Case of the above output with the following parameters:
1. Search for "Change Case"
2. Set parameters:
      - From: output of Extract Dataset tool
      - Change case of columns: c1
      - Delimited by: Tab
      - To: Upper case
3. Rename the output to Gene IDs and length


#### Running GOSeq

Steps: 
1. Search for the "GOSeq" tool
2. Set the parameters:
- Differentially expressed genes file: Gene IDs and differential expression
- Gene lengths file: Gene IDs and length
- Gene categories: Use a category file from history
- Gene category file: GO categories
- Output Options:
      - Output Top GO terms plot?: Yes
      - Extract the DE genes for the categories (GO/KEGG terms)?: Yes
3. Run the tool

GOSeq outputs include: 

1. A table (Ranked category list - Wallenius method) with the following columns for each GO term
2. A graph with the top 10 over-represented GO terms
3. A table with the differentially expressed genes (from the list we provided) associated to the GO terms (DE genes for categories (GO/KEGG terms))

### Building Workflows 

#### Creating a workflow 

#### Editing the workflow

#### Sharing your workflow

One of the most important benefits of using the Galaxy interface is that it enables users to share their workflows and histories with others. 